src/platform/i386/boot_comp.c

```diff
@@ -92,18 +95,19 @@ kern_boot_thd(struct captbl *ct, void *thd_mem, void *tcap_mem)
 	struct cap_pgtbl *         cap_pt;
 	pgtbl_t                    pgtbl;
 
+	assert(cos_info->cpuid == cpu_id);
 	assert(sizeof(struct cos_cpu_local_info) == STK_INFO_SZ);
 	memset(cos_info, 0, sizeof(struct cos_cpu_local_info));
-	cos_info->cpuid          = 0;
+	cos_info->cpuid          = cpu_id;
 	cos_info->invstk_top     = 0;
 	cos_info->overflow_check = 0xDEADBEEF;
-
-	ret = thd_activate(ct, BOOT_CAPTBL_SELF_CT, BOOT_CAPTBL_SELF_INITTHD_BASE, thd_mem, BOOT_CAPTBL_SELF_COMP, 0);
+	ret = thd_activate(ct, BOOT_CAPTBL_SELF_CT, BOOT_CAPTBL_SELF_INITTHD_BASE + cpu_id * captbl_idsize(CAP_THD), thd_mem, BOOT_CAPTBL_SELF_COMP, 0);
```

Comment:
I think we need wrappers to get the capability offset of a core id given the offset of the current core id and the capability type.

Reply:
Because we're going to access cap offsets of a core in many places including user-level components, like `llbooter`, `capmgr`, etc.

Reply:
good point, I will check this out.